### PR TITLE
Fix ignored Workspace generation when Project exists on the same directory

### LINKED ISSUE #4206: Workspace.swift ignored when Project.swift with same name is in the same directory
**Describe the bug**
Running `tuist generate` within a directory that contains both a `Workspace.swift` and a `Project.swift`, where both the workspace and project have the same name, causes the `Workspace.swift` to be ignored.

Previously (in Tuist 2.7.2) the `Workspace.swift` would replace the autogenerated one from the project.

**To Reproduce**
I've put together a basic example to both reproduce this issue as well as demonstrate that 2.7.2 did not have the same issue:

https://github.com/stuaustin/tuist-workspace-bug

1. cd into `3.0.0-bug`
2. run `tuist generate`
3. verify in Xcode that the workspace is missing the scheme "CustomScheme" described in the `Workspace.swift`.

**Expected behavior**
The workspace should be generated from `Workspace.swift`, and the scheme described in the `Workspace.swift` should be included in the resultant xcworkspace.

**Desktop (please complete the following information):**
- Tuist 3.0.0
- macOS 11.6.2
- Xcode 13.2.1

### PR DESCRIPTION
**Pull request by adellibovi**
Resolves #4206

### Short description 📝

This PR resolves a conflicts where Workspace was ignored when having a Project with the same name.

### How to test the changes locally 🧐

There is an example in #4206

### Checklist ✅

- [x] The code architecture and patterns are consistent with the rest of the codebase.
- [x] The changes have been tested following the [documented guidelines](https://docs.tuist.io/contributors/testing-strategy/).
- [x] The `CHANGELOG.md` has been updated to reflect the changes. In case of a breaking change, it's been flagged as such.
- [x] In case the PR introduces changes that affect users, the documentation has been updated.
- [x] In case the PR introduces changes that affect how the cache artifact is generated starting from the `TuistGraph.Target`, the `Constants.cacheVersion` has been updated.

### CODE REVIEW

File: Tests/TuistLoaderTests/Loaders/RecursiveManifestLoaderTests.swift [hunk 1]
```diff
@@ -258,6 +258,36 @@ final class RecursiveManifestLoaderTests: TuistUnitTestCase {
...
+
+        // When
+        let manifests = try subject.loadWorkspace(at: path.appending(RelativePath("Some/Path")))
+
+        // Then
+        XCTAssertEqual(manifests.path, path.appending(RelativePath("Some/Path")))
+        XCTAssertEqual(manifests.workspace, workspace)
+        XCTAssertEqual(withRelativePaths(manifests.projects), [
+            "Some/Path": projectA,
+        ])
```
Comment (danyf90) [inline Tests/TuistLoaderTests/Loaders/RecursiveManifestLoaderTests.swift:288]: What would have been the result with the old logic?

**Review by danyf90 — APPROVED** [review]

File: Tests/TuistLoaderTests/Loaders/RecursiveManifestLoaderTests.swift [hunk 2]
```diff
@@ -258,6 +258,36 @@ final class RecursiveManifestLoaderTests: TuistUnitTestCase {
...
     }
 
+    func test_loadWorkspace_withSameProjectName() throws {
+        // Given
+        let workspace = Workspace.test(
+            name: "ProjectA",
+            projects: [
+                ".",
+            ]
+        )
```
Comment (kwridan) [inline Tests/TuistLoaderTests/Loaders/RecursiveManifestLoaderTests.swift:268]: If I followed along, the bug we wasn't so much to do the the name of the workspace / project being similar, rather tuist 3.0.1 was acceidentally ignoring the Workspace manifest all together if it resided in the same directory!! 😅

As such the following modification to the workspace to make it distinct would help catch this.

```suggestion
        let workspace = Workspace.test(
            name: "MyWorkspace",
            projects: [
                ".",
            ],
            schemes: [
                Scheme(name: "CustomWorkspaceScheme")
            ]
        )
```

**Review by kwridan — APPROVED** [review]
Thanks @adellibovi

File: Tests/TuistLoaderTests/Loaders/RecursiveManifestLoaderTests.swift (hunk 2 above)
Comment (danyf90) [reply Tests/TuistLoaderTests/Loaders/RecursiveManifestLoaderTests.swift:268]: That's my understanding as well, and the origin of [my question](https://github.com/tuist/tuist/pull/4236#discussion_r825309442) 🚀

Comment (adellibovi) [reply Tests/TuistLoaderTests/Loaders/RecursiveManifestLoaderTests.swift]: Right! The change in the Unit Test will make sure it's the actual "custom" workspace, thank you folks! I will merge once it pass @danyf90 @kwridan

**Comment by integer256** [comment]
@adellibovi Thanks for fixing this, when will it be released?

**Pull request merged**